### PR TITLE
fix(aws-ecs-patterns): only create an A record if LB is public

### LINKED ISSUE #6702: ARecord is created even with publicLoadBalancer is set to false in ApplicationLoadBalancedFargateService
Even when the 'publicLoadBalancer' is set to false, ApplicationLoadBalancedFargateService still creates an ARecord with loadbalancer as target.

I don't see any reason to have an ARecord when the loadbalancer is private.

### Reproduction Steps

        // CloudFormation Resources
        this.service = new ecsPatterns.ApplicationLoadBalancedFargateService(this, 'Service', {
            .......
            .......
            .......
            publicLoadBalancer: false,
            protocol: elbv2.ApplicationProtocol.HTTPS,
            domainName: `abc.com`,
            domainZone: serviceHostedZone,
            .......
            .......
            .......

        });

### Environment

  - **CLI Version      :**
  - **Framework Version:**
  - **OS               :**
  - **Language         :**

### Other

It doesn't make sense to create an ARecord which never gets resolved when the 'publicLoadBalancer' is set false. Ideally record creation can be optional or atleast do it only when publicLoadBalancer is set true.

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by hencrice**
Before this change, we unnecessarily create A record when the load balancer(s) is non public-facing. This commit fixes #6702.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by hencrice** [comment]
For the failed PR linkter check, I'm a bit confused on what kind of tests it's looking for. It only says:
```
Run ./.github/actions/prlinter
  with:
    check: MANDATORY_CHANGES
  env:
    GITHUB_TOKEN: ***
Creating authenticated GitHub Client
⌛  Fetching PR number 6895
⌛  Fetching files for PR number 6895
⌛  Validating...
##[error]Fixes must contain a change to a test file
```

**Comment by hencrice** [comment]
> For the failed PR linkter check, I'm a bit confused on what kind of tests it's looking for. It only says:
> 
> ```
> Run ./.github/actions/prlinter
>   with:
>     check: MANDATORY_CHANGES
>   env:
>     GITHUB_TOKEN: ***
> Creating authenticated GitHub Client
> ⌛  Fetching PR number 6895
> ⌛  Fetching files for PR number 6895
> ⌛  Validating...
> ##[error]Fixes must contain a change to a test file
> ```

Found it. I suppose it just wants me to at least update some tests: https://github.com/aws/aws-cdk/blob/master/tools/prlint/index.js#L59

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts [hunk 1]
```diff
@@ -351,21 +351,23 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
...
+          recordName: props.domainName,
+          target: RecordTarget.fromAlias(new LoadBalancerTarget(loadBalancer)),
+        });
 
-      domainName = record.domainName;
+        domainName = record.domainName;
+      }
     }
 
+    new cdk.CfnOutput(this, 'LoadBalancerDNS', { value: loadBalancer.loadBalancerDnsName });
```
Comment (iamhopaul123) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts]: Just out of curiosity: why making this change?

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts [hunk 2]
```diff
@@ -297,20 +297,23 @@ export abstract class NetworkMultipleTargetGroupsServiceBase extends Construct {
...
-        const lb = this.createLoadBalancer(lbProps.name, lbProps.publicLoadBalancer);
+        const internetFacing = lbProps.publicLoadBalancer !== undefined ? lbProps.publicLoadBalancer : true;
+        const lb = this.createLoadBalancer(lbProps.name, internetFacing);
         this.loadBalancers.push(lb);
         for (const listenerProps of lbProps.listeners) {
           const listener = this.createListener(listenerProps.name, lb, listenerProps.port || 80);
           this.listeners.push(listener);
         }
-        this.createDomainName(lb, lbProps.domainName, lbProps.domainZone);
+        if (internetFacing) {
```
Comment (iamhopaul123) [inline packages/@aws-cdk/aws-ecs-patterns/lib/base/network-multiple-target-groups-service-base.ts]: Thank you!

File: packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts (hunk 1 above)
Comment (hencrice) [reply packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts]: oops I guess this is not necessary. Will move these 2 lines back to where they were

**Pull request merged**